File: 2857-total-distance-traveled/2857-total-distance-traveled.py
class Solution:
    def distanceTraveled(self, mainTank: int, additionalTank: int) -> int:
        s=0
        while(mainTank):
            if(mainTank<5):
                s+=mainTank
                mainTank=0
            else:
                s+=5
                mainTank-=5
                if(additionalTank>0):
                    additionalTank-=1
                    mainTank+=1
        return 10*s


        

File: 2857-total-distance-traveled/2857-total-distance-traveled.cpp
class Solution {
public:
    int distanceTraveled(int mainTank, int additionalTank) {
        int s=0;
        while(mainTank>0){
            if(mainTank<5){
                s+=mainTank;
                mainTank=0;
            }
            else{
                s+=5;
                mainTank-=5;
                if (additionalTank > 0) {
                    mainTank++;
                    additionalTank--; 
                    }
            }

        }
        int km=10*s;
        return km;
        
    }
};